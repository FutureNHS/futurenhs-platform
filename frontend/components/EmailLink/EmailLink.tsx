import React from "react";

interface Props {
  emailAddress: string;
}

const EmailLink = ({ emailAddress }: Props) => (
  <a
    href={`mailto:${encodeURI(emailAddress)}`}
    target="_blank"
    rel="noreferrer"
  >
    {emailAddress}
  </a>
);

export default EmailLink;
