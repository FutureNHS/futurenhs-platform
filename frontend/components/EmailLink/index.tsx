import EmailLink from "./EmailLink";

export { EmailLink };
