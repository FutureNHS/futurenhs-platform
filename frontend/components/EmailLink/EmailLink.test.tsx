import React from "react";

import { render } from "@testing-library/react";

import EmailLink from "./EmailLink";

describe(EmailLink, () => {
  it("renders correctly", () => {
    const container = render(
      <EmailLink emailAddress="my-email-address@email.com" />
    );

    expect(container.asFragment()).toMatchSnapshot();
  });
});
