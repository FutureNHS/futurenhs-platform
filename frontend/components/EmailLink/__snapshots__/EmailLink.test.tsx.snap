// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailLink renders correctly 1`] = `
<DocumentFragment>
  <a
    href="mailto:my-email-address@email.com"
    rel="noreferrer"
    target="_blank"
  >
    my-email-address@email.com
  </a>
</DocumentFragment>
`;
